src/beanmachine/ppl/diagnostics/tools/marginal1d/tool.py

```diff
@@ -0,0 +1,172 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""Marginal 1D diagnostic tool for a Bean Machine model."""
+
+from typing import TypeVar
+
+from beanmachine.ppl.diagnostics.tools.marginal1d import utils
+from beanmachine.ppl.diagnostics.tools.utils.base import Base
+from beanmachine.ppl.inference.monte_carlo_samples import MonteCarloSamples
+from bokeh.models.callbacks import CustomJS
+
+
+T = TypeVar("T", bound="Marginal1d")
+
+
+class Marginal1d(Base):
+    """Marginal (1D) tool.
+
+    Parameters
+    ----------
+    mcs : MonteCarloSamples
```

Comment:
Can we take an `InferenceData` here and pass `mcs.to_inference_data()` at callsites? It would reduce effort to upstream to arviz

Reply:
See my response below on your comment in the serializer.